sci-libs/vtk/vtk-9.1.0.ebuild

```diff
@@ -0,0 +1,601 @@
+	if use postgres; then
+		mycmakeargs+=(
+			-DVTK_MODULE_ENABLE_VTK_IOPostgreSQL="WANT"
+			-DVTK_MODULE_ENABLE_VTK_IOSQL="WANT"
+		)
+	fi
+
+	if use python; then
+		mycmakeargs+=(
+			-DVTK_ENABLE_WRAPPING=ON
+			-DPython3_EXECUTABLE="${PYTHON}"
+			-DVTK_PYTHON_SITE_PACKAGES_SUFFIX="lib/${EPYTHON}/site-packages"
+		)
+	fi
+
+	if use qt5; then
+		mycmakeargs+=( -DVTK_MODULE_ENABLE_VTK_GUISupportQt="WANT" )
+		if use mysql || use postgres; then
+			mycmakeargs+=( -DVTK_MODULE_ENABLE_VTK_GUISupportQtSQL="WANT" )
+		fi
+		if use rendering; then
+			mycmakeargs+=( -DVTK_MODULE_ENABLE_VTK_RenderingQt="WANT" )
+		fi
+		if use views; then
+			mycmakeargs+=( -DVTK_MODULE_ENABLE_VTK_ViewsQt="WANT" )
+		fi
+	fi
+
+	if use rendering || use web || use all-modules; then
+		# needs patched version
+		mycmakeargs+=( -DVTK_MODULE_USE_EXTERNAL_VTK_libharu=OFF )
+	fi
+
+	if use rendering; then
+		mycmakeargs+=(
+			-DVTK_MODULE_ENABLE_VTK_IOExportGL2PS="WANT"
+			-DVTK_USE_SDL2=ON
+# FIXME: this should be default with VTK_USE_EXTERNAL=ON. Verify this!
+#			-DVTK_MODULE_USE_EXTERNAL_VTK_gl2ps=ON
+#			-DVTK_MODULE_USE_EXTERNAL_VTK_glew=ON
+#			-DVTK_MODULE_USE_EXTERNAL_VTK_libproj=ON
+		)
+	fi
+
+	# FIXME: needs improvements (and has been improved upstream by providing
+	# the VTK{,Large}DataFiles packages. Still need to thoroughly test
+	# if we need to download additional files.
+	if use test; then
+		ewarn "Testing requires VTK_FORBID_DOWNLOADS=OFF by upstream."
+		ewarn "Care has been taken to pre-download all required files."
+		ewarn "In case you find missing files, please inform me."
+		mycmakeargs+=(
+			-DVTK_BUILD_TESTING=ON
+			-DVTK_DATA_EXCLUDE_FROM_ALL=ON
+			-DVTK_FORBID_DOWNLOADS=OFF
+		)
+	else
+		mycmakeargs+=(
+			-DVTK_BUILD_TESTING=OFF
+			-DVTK_FORBID_DOWNLOADS=ON
+		)
+	fi
+
+	# TODO: remote the USE flag and enable by default
+	if use theora; then
+		mycmakeargs+=( -DVTK_MODULE_ENABLE_VTK_IOOggTheora="WANT" )
+	fi
+
+	# FIXME: upstream provides 4 threading models, as of 9.1.0. These are
+	# sequential, stdpthread, openmp and tbb. AFAICS all of them can be
+	# enabled at the same time. Sequential and Stdthread are enabled by
+	# default. The default selected type for the build is sequential.
+	# Assuming sequential < stdpthread < openmp < tbb wrt speed, although
+	# this is dependent on the actual scenario where threading is used.
+	if use tbb; then
+		mycmakeargs+=( -DVTK_SMP_IMPLEMENTATION_TYPE="TBB" )
+	elif use openmp; then
+		mycmakeargs+=( -DVTK_SMP_IMPLEMENTATION_TYPE="OpenMP" )
+	elif use threads; then
+		mycmakeargs+=( -DVTK_SMP_IMPLEMENTATION_TYPE="STDThread" )
+	else
+		mycmakeargs+=( -DVTK_SMP_IMPLEMENTATION_TYPE="Sequential" )
+	fi
+
+	if use openmp; then
+		mycmakeargs+=( -DVTK_SMP_ENABLE_OPENMP=ON )
+	fi
+
+	if use tbb; then
+		mycmakeargs+=( -DVTK_SMP_ENABLE_TBB=ON )
+	fi
+
+	# ON by default, hence the else clause is neede
+	if use threads; then
+		mycmakeargs+=( -DVTK_SMP_ENABLE_STDTHREAD=ON )
```

Comment:
just =$(usex threads)? no need for the if statement?

Reply:
Thought about this at one time as well. Should do it.

Reply:
If wanted to, can even use it for multiple flags and potentially use a single'ish array, e.g.
```
array=(
	-DSTUFF
	$(usex flag '
		-DARG
		-DARG2' '')
	$(usex flag2 '
		-DARG3
		-DARG4' '')
)
```
But whether this is better is debatable (either way, note to replace by `$(usev flag -DFLAG)` whenever this goes EAPI-8 given that extra `''` is always ugly).

Reply:
Thanks for your feedback @ionenwks. There's only really a few which can be eliminated without causing cmake warnings about used but undefined variables. Most if statements are there, because the variables will only be defined once a certain scope is enabled.

Reply:
That form of `$(usex flag 'add this or nothing' '')` or eapi8 `$(usev flag this-or-nothing)` doesn't set the variable at all if the flag isn't, it's essentially an embed if statement